lib/Transforms/TensorToScalars/TensorToScalars.cpp

```diff
@@ -44,12 +76,18 @@ static SmallVector<Value> buildExtractOps(OpBuilder &builder,
 
   // Create extract ops in "natural" order (dimension-by-dimension)
   SmallVector<Value> values;
-  for (auto dim : inputType.getShape()) {
-    for (int i = 0; i < dim; ++i) {
-      Value index = builder.create<arith::ConstantIndexOp>(loc, i);
-      Value element = builder.create<tensor::ExtractOp>(loc, input, index);
-      values.push_back(element);
-    }
+  ImplicitLocOpBuilder b(loc, builder);
+  const auto *maxDimension = llvm::max_element(inputType.getShape());
+  SmallVector<Value> constVals;
+  for (auto i = 0; i < *maxDimension; ++i) {
+    constVals.push_back(b.create<arith::ConstantIndexOp>(i));
+  }
+  auto rawIndices = getAllIndices(inputType);
+  for (SmallVector<int64_t> indices : rawIndices) {
+    SmallVector<Value> idxRange = llvm::to_vector(
```

Comment:
It seems like the bug here was about incorrect handling of index ranges, but the test isn't exercising that? As a test, I reverted these changes to the pattern but kept the new patterns for `tensor.extract` and `arith.constant` and the test still seemed to pass?

Reply:
That is so weird :o When I was first testing this pass with 1x1xi8 I was getting an incorrect number of results because the type converter in that change was extracting 0 twice and it should have been just one element.

now i'm confused why it's passing...

Reply:
Can you try the same check on your end? Maybe I simply didn't give vscode enough time to write file changes back to disk and the test does actually fail?

Reply:
I think once I added in the extract and constant patterns then the code that you're reverting doesn't even make a difference. Change that diff to
```
  // Create extract ops in "natural" order (dimension-by-dimension)
  SmallVector<Value> values;
  return values;
```

and you'll see it passes too.